botocore/parsers.py

```diff
@@ -598,7 +598,7 @@ def _do_error_parse(self, response, shape):
             status_message = http.client.responses.get(
                 response['status_code'], ''
             )
-            if status_message:
+            if status_message and hasattr(e, 'add_note'):
```

Comment:
I think this requires Python 3.11+ so this is probably not the approach we want to take to make sure this works everywhere.